esp/public/media/scripts/ajaxschedulingmodule/ESP/Sections.js

```diff
@@ -541,6 +544,136 @@ function Sections(sections_data, section_details_data, categories_data, teacher_
+        var new_timeslots1 = this.matrix.timeslots.get_timeslots_to_schedule_section(section1, old_timeslots2[0]);
+        var new_timeslots2 = this.matrix.timeslots.get_timeslots_to_schedule_section(section2, old_timeslots1[0]);
+
+        // Abort if there isn't enough time for either section in their new classroom
+        if(new_timeslots1 === null){
+            this.matrix.messagePanel.addMessage("Error: not enough time to swap the first section to the new room.", color = "red");
+            this.unselectSection();
+            return;
+        }
+        if(new_timeslots2 === null){
+            this.matrix.messagePanel.addMessage("Error: not enough time to swap the second section to the new room.", color = "red");
+            this.unselectSection();
+            return;
+        }
+
+        // Get the sections you'll be swapping with (might be different than section1/section2)
+        var swapping_sections1 = new_timeslots2.map(ts => this.matrix.getCell(room1, ts).section);
+        var swapping_sections2 = new_timeslots1.map(ts => this.matrix.getCell(room2, ts).section);
+        // If there's more than one swapping section, confirm the user is OK with this
+        if(_.uniq(swapping_sections1.filter(Boolean)).length > 1 && !confirm("More than one section will need to be swapped from the old room. Is this OK?")){
+            return;
+        } else if(_.uniq(swapping_sections2.filter(Boolean)).length > 1 && !confirm("More than one section will need to be swapped from the new room. Is this OK?")){
+            return;
+        }
+        $j(".ui-tooltip").hide() // There seems to be a bug or something caused by the confirm box that causes the tooltip to get stuck
+        var old_assignments1 = _.uniq(swapping_sections1.filter(Boolean)).map(sec => ({"section": sec, "timeslots": this.scheduleAssignments[sec.id].timeslots, "room_id": room1}));
+        var old_assignments2 = _.uniq(swapping_sections2.filter(Boolean)).map(sec => ({"section": sec, "timeslots": this.scheduleAssignments[sec.id].timeslots, "room_id": room2}));
+        var ignore_sections = _.uniq(_.union(swapping_sections1, swapping_sections2));
+
+        // Determine new schedule for room 1 sections and validate assignments
+        var new_assignments1 = [];
+        var start_slot = this.matrix.timeslots.get_by_id(old_timeslots2[0]);
+        for(var sec of swapping_sections1){
+            if(sec){
+                if(new_assignments1.length == 0 || sec !== new_assignments1[new_assignments1.length - 1].section){
+                    var new_timeslots = this.matrix.timeslots.get_timeslots_to_schedule_section(sec, start_slot.id);
+                    var valid = this.matrix.validateAssignment(sec, room2, new_timeslots, ignore_sections);
+                    if(!valid.valid){
+                        console.log(valid.reason);
+                        this.matrix.messagePanel.addMessage("Error: can not swap those classes.", color = "red");
```

Comment:
It would be more helpful to give an indication as to why the classes can not be swapped